test/controllers/WhomToPayControllerSpec.scala

```diff
@@ -56,6 +56,60 @@ class WhomToPayControllerSpec extends SpecBase with MockitoSugar {
       application.stop()
     }
 
+    "removes bank details if answer changes" in {
+
+      val userAnswers = UserAnswers(userIdentification).set(WhomToPayPage, WhomToPay.Representative).success.value
+
+      userAnswers.set(BankDetailsPage, BankDetails("Natural Numbers Inc", "123456", "12345678"))
+
+      when(mockSessionRepository.set(any())) thenReturn Future.successful(true)
+
+      val application =
+        applicationBuilder(userAnswers = Some(userAnswers))
+          .build()
+
+      val request =
+        FakeRequest(POST, whomToPayRoute)
+          .withFormUrlEncodedBody(("value", WhomToPay.options(form).head.value.get))
+
+      val result = route(application, request).value
+
+      userAnswers.get(BankDetailsPage) mustBe None
```

Comment:
This will always be None - regardless of your changes.  
`userAnswers.set(BankDetailsPage, BankDetails("Natural Numbers Inc", "123456", "12345678"))` does not update the userAnswers - it returns a new one (which is then discarded).

I will try to dig out an example of a test asserting the state of userAnswers.

Reply:
See `CreateOrAmendCaseControllerSpec` - "redirect to the next page when valid data is submitted"

Reply:
thank you

Reply:
Oh I didnt see tip above.